crates/runtime/src/traphandlers/macos.rs

```diff
@@ -409,13 +381,13 @@ unsafe fn handle_exception(request: &mut ExceptionRequest) -> bool {
 /// a native backtrace once we've switched back to the thread itself. After
 /// the backtrace is captured we can do the usual `longjmp` back to the source
 /// of the wasm code.
-unsafe extern "C" fn unwind(wasm_pc: *const u8, jmp_buf: *const u8) -> ! {
-    tls::with(|state| {
-        if let Some(state) = state {
-            state.capture_backtrace(wasm_pc);
-        }
+unsafe extern "C" fn unwind(wasm_pc: *const u8) -> ! {
+    let jmp_buf = tls::with(|state| {
+        let state = state.unwrap();
+        state.capture_backtrace(wasm_pc);
+        state.jmp_buf.get()
     });
-
+    debug_assert!(!jmp_buf.is_null());
```

Comment:
There was a check in the unix implementation that the `jmp_buf` could be set to 1 (meaning a custom signal handler has run and the trap handler should return early). Is it not necessary anymore?

Reply:
Ah yeah I realized that it was no longer necessary since we don't run the custom trap handler on macOS, it's only run on Unix and Windows now.